aopy/analysis/base.py

```diff
@@ -2372,4 +2376,261 @@ def simulate_ideal_trajectories(targets, origin=[0.0, 0.0, 0.0], resolution=1000
+
+            plt.subplot(1,3,2)
+            im = aopy.visualization.plot_spatial_drive_map(dists_mean, 
+                                            elec_data=True, cmap='Grays')
+            im.set_clim(*clim)
+            plt.axis('off')
+            plt.colorbar(im, shrink=0.5)
+            plt.title('dist means')
+
+            plt.subplot(1,3,3)
+            im = aopy.visualization.plot_spatial_drive_map(dists_std, 
+                                            elec_data=True, cmap='Grays')
+            plt.axis('off')
+            plt.colorbar(im, shrink=0.5)
+            plt.title('dist std')
+
+        .. image:: _images/calc_statistic_random_trials.png
+    '''
+    all_trials = np.arange(len(data))
+    np.random.shuffle(all_trials)
+    n_div = len(data)//n_trials
+    if n_div < 1:
+        raise ValueError(f"Not enough trials in data ({len(data)}) to perform the calculation with n_trials ({n_trials}).")
+    trials = np.reshape(all_trials[:n_div*n_trials], (n_div, n_trials))
+
+    dist = []
+    for k in range(n_div):
+        try:
+            dist.append(statistic(data[trials[k]]))
+        except KeyError:
+            dist.append(statistic(data.iloc[trials[k]].reset_index(drop=True)))
+
+    return np.array(dist)
+
+def _calc_statistic_random_trials(args):
+    '''
+    Helper function to unpack arguments for parallel processing.
+    '''
+    return calc_statistic_random_trials(*args)
+
+def calc_trial_bootstraps(data, n_trials=300, n_bootstraps=30, 
+                          statistic=partial(np.mean, axis=0), parallel=False):
+    '''
+    Repeatedly call :func:`~aopy.analysis.calc_statistic_random_trials` to generate
+    statistics over groups of trials.
+
+    Args:
+        data (ntrials, nch): data to calculate the statistic on
+        n_trials (int): number of trials to use in each bootstrap
+        n_bootstraps (int): number of bootstrap trials to perform
+        statistic (function): function to calculate the statistic on the data. Should
+            take the form `f(x) = y` where `x` is a 2D array of shape (n_trials, nch)
+            and `y` is a 1D array of shape (nch). Default is np.mean(x, axis=0).
+        parallel (bool or mp.pool.Pool, optional): Whether to run the bootstraps in parallel.
+            If True, uses a multiprocessing pool with 10 workers. If a Pool is provided,
+            it will use that pool instead. If False, runs the bootstraps sequentially.
+
+    Returns:
+        (n_bootstraps, len(data)//n_trials, nch) array: multiple bootstraps of distributions
+            of the statistic applied to divisions of the data.
+    '''
+    assert n_bootstraps > 0, "must have at least 1 bootstrap"
+    assert n_trials > 0, "must have at least 1 trial"
+    assert len(data) >= n_trials, "data must have at least as many trials as n_trials"
+    assert callable(statistic), "statistic must be a callable function"
+
+    args_list = [(data, n_trials, statistic)] * n_bootstraps
```

Comment:
it seems like copying the data n_bootstraps times could end up taking up quite a bit of RAM since the data doesn't change. I see its necessary for parallel processing but I don't think it should be used for sequential processing. That way you can use the one advantage of sequential processing, low RAM